### ISSUE TITLE
colexec: a drift in memory accounting (probably with flat bytes)

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#45425]
While working on #45360, I ran into a situation such that we were trying to shrink the memory account when calling `ReleaseBatch` by the bigger amount than the account had registered as "used". This could be easily reproduced with:
```
# LogicTest: fakedist-vec

statement ok
CREATE TABLE kv (
  k INT PRIMARY KEY,
  v INT,
  w INT,
  s STRING,
  i INTERVAL
)

statement ok
INSERT INTO kv VALUES
(1, 2, 3, 'a', '1min'),
(3, 4, 5, 'a', '2sec'),
(5, NULL, 5, NULL, NULL),
(6, 2, 3, 'b', '1ms'),
(7, 2, 2, 'b', '4 days'),
(8, 4, 2, 'A', '3 years')

query IT rowsort
SELECT count(*), kv.s FROM kv GROUP BY kv.s
----
1 A
1 NULL
2 a
2 b
```
The panic message `panic: hash-router-[2,4]-unlimited: no bytes in account to release, current 79872, free 79940` seems to suggest that we have an extra `[]byte` of 64 bytes plus 4 bytes for offset, but I couldn't pin point where the problem occurs. We should figure that out.

**Comment by pbardea** [comment]
I'm not sure if it's much help, but I saw this flake a couple of times on CI on my branch today. Once on (TestLogic/fakedist-disk/int_size/issue_32846)[https://teamcity.cockroachdb.com/viewLog.html?buildId=1816072&buildTypeId=Cockroach_UnitTests] and once on TestHashRouterRandom.

**Comment by yuzefovich** [comment]
No, that was different issue, and it was fixed by #46220.
